src/dll/Hooks/AssertionFailed.cpp

```diff
@@ -0,0 +1,85 @@
+#include "AssertionFailed.hpp"
+#include "Addresses.hpp"
+#include "App.hpp"
+#include "Hook.hpp"
+#include "stdafx.hpp"
+
+namespace
+{
+
+bool isAttached = false;
+
+/// @pattern 4C 89 4C 24 20 53 55 56 57 41 54 41 56 48 83 EC 68 80 3D ? ? ? 00 00 49 8B F8 8B F2 48 8B E9
+__int64 _AssertionFailed(const char*, int, const char*, const char*...);
+
+Hook<decltype(&_AssertionFailed)> AssertionFailed_fnc(4285205681U, &_AssertionFailed);
+
+__int64 _AssertionFailed(const char* aFile, int aLineNum, const char* aCondition, const char* aMessage...)
+{
+    va_list args;
+
+    va_start(args, aMessage);
+    spdlog::error("Crash report");
+    spdlog::error("------------");
+    spdlog::error("File: {}", aFile);
+    spdlog::error("Line: {}", aLineNum);
+    if (aCondition)
+    {
+        spdlog::error("Condition: {}", aCondition);
+    }
+    if (aMessage)
+    {
+        char buffer[1024];
```

Comment:
Is the game also limiting this message to 1024 characters? If not, we can make it longer.

Reply:
Yes, I'll add a comment about that and change 1024 to 0x400, easier to match it when reversing.